synapseclient/api/file_services.py

```diff
@@ -0,0 +1,335 @@
+    )
+    return await client.rest_post_async(
+        uri,
+        json.dumps(body),
+        endpoint=client.fileHandleEndpoint,
+    )
+
+
+async def post_external_object_store_filehandle(
+    s3_file_key: str,
+    file_path: str,
+    storage_location_id: int,
+    mimetype: str = None,
+    md5: str = None,
+    synapse_client: Optional["Synapse"] = None,
+) -> Dict[str, Union[str, int]]:
+    """
+    Create a new FileHandle representing an external object.
+    <https://rest-docs.synapse.org/rest/POST/externalFileHandle.html>
+
+    Arguments:
+        s3_file_key:         S3 key of the uploaded object
+        file_path:           The local path of the uploaded file
+        storage_location_id: The optional storage location descriptor
+        mimetype:            The Mimetype of the file, if known.
+        md5:                 The file's content MD5, if known.
+        synapse_client: If not passed in or None this will use the last client from
+            the `.login()` method.
+
+    Returns:
+        A FileHandle for objects that are stored externally.
+        <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/file/ExternalFileHandleInterface.html>
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+    if mimetype is None:
+        mimetype, _ = mimetypes.guess_type(file_path, strict=False)
+    file_handle = {
+        "concreteType": concrete_types.EXTERNAL_OBJECT_STORE_FILE_HANDLE,
+        "fileKey": s3_file_key,
+        "fileName": os.path.basename(file_path),
+        "contentMd5": md5 or utils.md5_for_file(file_path).hexdigest(),
+        "contentSize": os.stat(file_path).st_size,
+        "storageLocationId": storage_location_id,
+        "contentType": mimetype,
+    }
+
+    return await client.rest_post_async(
+        "/externalFileHandle", json.dumps(file_handle), client.fileHandleEndpoint
+    )
+
+
+async def post_external_filehandle(
+    external_url: str,
+    mimetype: str = None,
+    md5: str = None,
+    file_size: int = None,
+    synapse_client: Optional["Synapse"] = None,
+) -> Dict[str, Union[str, int]]:
+    """
+    Create a new FileHandle representing an external object.
+    <https://rest-docs.synapse.org/rest/POST/externalFileHandle.html>
+
+    Arguments:
+        externalURL:  An external URL
+        mimetype:     The Mimetype of the file, if known.
+        md5:          The file's content MD5.
+        file_size:    The size of the file in bytes.
+        synapse_client: If not passed in or None this will use the last client from
+            the `.login()` method.
+
+    Returns:
+        A FileHandle for objects that are stored externally.
+        <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/file/ExternalFileHandleInterface.html>
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+    file_name = external_url.split("/")[-1]
```

Comment:
You can use `os.path.basename(external_url)` for this.

<img width="575" alt="image" src="https://github.com/Sage-Bionetworks/synapsePythonClient/assets/32107699/18583ab8-87da-4a7e-84b4-77808c9fa578">